### PR TITLE
Add D3 for ui charts.

### PR DESCRIPTION
**Pull request by BramGruneir**
This just adds the D3 library and typescript typings for it.

### CODE REVIEW

**Comment by tamird** [comment]
I think you forgot to check in `d3.min.js` itself?

**Comment by BramGruneir** [comment]
It was a gitignore issue.  Fixed and added the file.

**Comment by tamird** [comment]
LG

File: .gitignore [hunk 1]
```diff
@@ -15,6 +15,9 @@ certs
 *.[568vq]
 [568vq].out
 
+# Override the *.[568vq] or else directories like x.x.5/ will be excluded.
```
Comment (bdarnell) [inline .gitignore]: Just remove the rule above. We picked it up from github's default .gitignore for go projects, but I don't think we really need it. It matches intermediate files created by older versions of the go compiler (or at least I've never seen them in practice).

**Comment by bdarnell** [comment]
LGTM

File: .gitignore (hunk 1 above)
Comment (BramGruneir) [inline .gitignore]: Done.

**Pull request merged**